Index: bob_quilt/CMakeLists.txt
===================================================================
--- bob_quilt.orig/CMakeLists.txt	2013-01-15 17:02:24.386696615 +0100
+++ bob_quilt/CMakeLists.txt	2013-01-15 17:03:43.895037524 +0100
@@ -76,7 +76,7 @@
   set(COMMON_CXX_FLAGS "-std=gnu++0x")
   set(COMMON_C_FLAGS "-std=gnu99")
 else()
-  set(COMMON_CXX_FLAGS "-std=c++0x")
+  set(COMMON_CXX_FLAGS "-std=c++98")
   set(COMMON_C_FLAGS "-std=c99")
 endif()
 
